libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -401,6 +401,25 @@ bool AP_DDS_Client::start(void)
     return true;
 }
 
+// read function triggered at every subscription callback
+void AP_DDS_Client::on_topic (uxrSession* uxr_session, uxrObjectId object_id, uint16_t request_id, uxrStreamId stream_id, struct ucdrBuffer* ub, uint16_t length, void* args)
+{
+    (void) uxr_session; (void) object_id; (void) request_id; (void) stream_id; (void) length;
+    /*
+    TEMPLATE for reading to the subscribed topics
+    1) Store the read contents into the ucdr buffer
+    2) Deserialize the said contents into the topic instance
+    */
+    sensor_msgs_msg_Joy topic;
+    sensor_msgs_msg_Joy_deserialize_topic(ub, &topic);
+
+    uint32_t* count_ptr = (uint32_t*) args;
+    (*count_ptr)++;
+    
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Receivied sensor_msgs/Joy: %f, %f, %f, %f",
+        topic.axes[0], topic.axes[1], topic.axes[2], topic.axes[3]);
```

Comment:
Let's just print the array length. If someone sends a Joy message with less than 4 fields, it will cause a runtime crash.

Reply:
Thanks for this  ! Have made the said changes